main/src/ca/uwaterloo/flix/language/phase/TypeReconstruction.scala

```diff
@@ -417,10 +417,11 @@ object TypeReconstruction {
       val eff = Type.mkUnion(eff1 :: es.map(_.eff), loc)
       TypedAst.Expr.Do(op, es, tpe, eff, loc)
 
-    case KindedAst.Expr.InvokeMethod2(exp, name, exps, tvar, evar, loc) =>
+    case KindedAst.Expr.InvokeMethod2(exp, name, exps, mvar, tvar, evar, loc) =>
       val e = visitExp(exp)
-      val es = exps.map(visitExp(_))
-      TypedAst.Expr.InvokeMethod2(e, name, es, subst(tvar), subst(evar), loc)
+      val es = exps.map(visitExp)
+      // val t = Type.Cst(TypeConstructor.JvmMethod(name.name, es.length), loc) = subst(mvar)
+      TypedAst.Expr.InvokeMethod2(e, name, es, subst(mvar), subst(tvar), subst(evar), loc)
```

Comment:
Here the cool stuff happens! Do:

```scala
val method = subst(mvar) match {
    case Type.Cst(TypeConstructor.Method(m) => m
    case _ => throw InternalCompiler...
}
```

Then you can create an InvokeMethod using m and subst(tvar) and subst(evar).

(Note InvokeMethod-- not InvokeMethod2, there will no longer be such a 2 on TypedAst).